### PR TITLE
feature: RTL support added

### PR DESCRIPTION
**Pull request by alamgircsebd**

### CODE REVIEW

**Comment by diegoversiani** [comment]
@alamgircsebd, thank you for improving the WooCommece checks and activation links.

There are a few general issues with this PR:

- the style files (`.scss`) should always be split into partial files
- compiled `css/*.css` and `js/*.js` files need to be committed, they are ignored during the PR review, however, they are needed for the plugin to work properly.
- pixel values need to be wrapped with the `remsize()` sass function, this allows theme developers to rebuild the styles using `rem` size units if needed.

**Comment by diegoversiani** [comment]
@alamgircsebd, I think we can change some styles definitions from `left` to `right` and vice versa using SASS value interpolation.

This could potentially avoid the creation of new RTL files for each partial, for each theme or plugin we have compatibility with.

For example, instead of this as a separate RTL file:

```scss
form.woocommerce-checkout .fc-wrapper #order_review table.woocommerce-checkout-review-order-table .product-quantity {
    right: 65px;
    left: auto;
    transform: translateX(100%) translateY(-25%);
}
```

We could do this in the original partial file:

```scss
form.woocommerce-checkout .fc-wrapper #order_review table.woocommerce-checkout-review-order-table .product-quantity {
    #{ $horizontal-position }: 65px;
    transform: translateX(100%) translateY(-25%);
}
```

For that, we need to identify the CSS properties that change from LTR to RTL and change all original partial files to support the new variables.

Then, to generate separate styles for RTL, we can import all the same partials but change the variables.

**Comment by alamgircsebd** [comment]
> @alamgircsebd, I think we can change some styles definitions from `left` to `right` and vice versa using SASS value interpolation.
> 
> This could potentially avoid the creation of new RTL files for each partial, for each theme or plugin we have compatibility with.
> 
> For example, instead of this as a separate RTL file:
> 
> ```css-scss
> form.woocommerce-checkout .fc-wrapper #order_review table.woocommerce-checkout-review-order-table .product-quantity {
>     right: 65px;
>     left: auto;
>     transform: translateX(100%) translateY(-25%);
> }
> ```
> 
> We could do this in the original partial file:
> 
> ```css-scss
> form.woocommerce-checkout .fc-wrapper #order_review table.woocommerce-checkout-review-order-table .product-quantity {
>     #{ $horizontal-position }: 65px;
>     transform: translateX(100%) translateY(-25%);
> }
> ```
> 
> For that, we need to identify the CSS properties that change from LTR to RTL and change all original partial files to support the new variables.
> 
> Then, to generate separate styles for RTL, we can import all the same partials but change the variables.

please review now

File: sass/checkout-layout-rtl.scss [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+.has-checkout-layout--multi-step .fc-step__substep-text {
+    padding-left: 0;
+    padding-right: 80px;
+    width: calc( 100% - 230px );
+}
+
+div.woocommerce form .form-row.woocommerce-validated:not([class*=woocommerce-invalid]):not(.fc-no-validation-icon) .select2-selection:after,
+div.woocommerce form .form-row.woocommerce-validated:not([class*=woocommerce-invalid]):not(.fc-no-validation-icon):not(.fc-select2-field) .woocommerce-input-wrapper:after {
+    left: 20px;
+    right: auto !important;
```
Comment (diegoversiani) [inline sass/checkout-layout-rtl.scss]: @alamgircsebd High level `.scss` files should not contain code that generates CSS directly, only imports for necessary abstract files and partial files are permitted.

Move the new code to the appropriate existing partial files, or create new partial files for it.

File: sass/checkout-layout-rtl.scss [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+// ------------------------------------
+// IF NOT EXITS WITH VARIABLES
+// ------------------------------------
+body.woocommerce-checkout form.woocommerce-checkout .fc-wrapper .fc-step__substep-fields,
+div.woocommerce .fc-expansible-form-section .expansible-section__toggle-plus{
+	text-align: $checkout-rtl-alignment-left;
+}
+
+.has-checkout-layout--multi-step.woocommerce-checkout form.woocommerce-checkout .fc-wrapper [data-step-complete] .fc-step__substep .fc-step__substep-title:before {
+    right: 0px;
```
Comment (diegoversiani) [inline sass/checkout-layout-rtl.scss]: @alamgircsebd avoid the use of size unit when value is `0`.

File: sass/checkout-layout-rtl.scss (hunk 1 above)
Comment (diegoversiani) [inline sass/checkout-layout-rtl.scss]: @alamgircsebd Replace `px` with `remsize()`, and use new alignment variables for this code.

File: sass/partials/_order-summary.scss [hunk 3]
```diff
@@ -162,18 +163,19 @@ form.woocommerce-checkout .fc-wrapper #order_review table.woocommerce-checkout-r
...
 form.woocommerce-checkout .fc-wrapper #order_review table.woocommerce-checkout-review-order-table img {
-	float: left;
+	float: $checkout-rtl-alignment-left;
 	margin-left: - #{ $order-summary-image-padding + $global-spacing-tiny };
 	padding: $order-summary-image-padding;
 	width: $order-summary-image-width !important; // Needed to override inline styles added by some themes
 	height: auto !important; // Needed to override inline styles added by some themes
 	background-color: $order-summary-image-background-color;
 	border: $order-summary-image-border;
+	margin-right: $wc-checkout-review-order-table-img;// -8px;
```
Comment (diegoversiani) [inline sass/partials/_order-summary.scss]: @alamgircsebd keep various `margin` properties together at the top.

File: sass/partials/_order-summary.scss [hunk 4]
```diff
@@ -183,8 +185,9 @@ form.woocommerce-checkout .fc-wrapper #order_review table.woocommerce-checkout-r
 }
 
 form.woocommerce-checkout .fc-wrapper #order_review table.woocommerce-checkout-review-order-table .product-details {
-	float: left;
-	padding-left: $global-spacing-small;
+	float: $checkout-rtl-alignment-left;
+	//padding-left: $global-spacing-small;
```
Comment (diegoversiani) [inline sass/partials/_order-summary.scss]: @alamgircsebd remove commented code.

**Review by diegoversiani — CHANGES_REQUESTED** [review]

**Comment by alamgircsebd** [comment]
Review now

**Comment by diegoversiani** [comment]
@alamgircsebd there are various layout issues with the RTL version, I'm using the default WooCommerce Storefront theme. See the images attached.

Screen Shot 2022-02-23 at 08 27 01
Screen Shot 2022-02-23 at 08 27 29
Screen Shot 2022-02-23 at 08 28 18
Screen Shot 2022-02-23 at 08 32 28
Screen Shot 2022-02-23 at 08 34 47
Screen Shot 2022-02-23 at 08 25 25
Screen Shot 2022-02-23 at 08 30 40
Screen Shot 2022-02-23 at 08 39 47

**Comment by alamgircsebd** [comment]
Review now

**Comment by diegoversiani** [comment]
@alamgircsebd There are still things that were not addressed, please check all the (pending) comments in the code "files changed" tab.

Also, check other layout issues in the screenshots below:

Screen Shot 2022-02-26 at 06 58 55
Screen Shot 2022-02-26 at 07 02 29
Screen Shot 2022-02-26 at 07 02 52
Screen Shot 2022-02-26 at 07 34 27
Screen Shot 2022-02-26 at 07 31 46
Screen Shot 2022-02-26 at 07 04 34

**Pull request merged**